game-app/game-core/src/main/java/games/strategy/triplea/delegate/battle/casualty/CasualtySelector.java

```diff
@@ -150,13 +150,16 @@ public static CasualtyDetails selectCasualties(
       killAmphibiousFirst(killed, sortedTargetsToPickFrom);
     }
 
-    // if partial retreat isn't possible, then ensure units with marine bonus are removed first or
-    // last depending on whether the bonus is negative or positive
+    // if partial retreat isn't possible and it was an amphibious assault, then ensure units with
+    // marine bonus are removed first or last depending on whether the bonus is negative or positive
```

Comment:
It would be proper to add a "respectively" here. Without it, it might be unclear that "negative" refers to "first" and "positive" refers to "last". Also, a mention of "amongst the same-named ones", referring to "units" may be added so to avoid that someone might think that, for example, the program will automatically take out infantry before offloaded marines, whereas here we are referring only to taking out non-offloaded marines before offloaded ones.